algo/src/main/java/com/hartwig/serve/sources/ckb/ActionableEvidenceFactory.java

```diff
@@ -202,15 +202,34 @@ static EvidenceLevel resolveLevel(@Nullable String evidenceLabel) {
 
     @Nullable
     @VisibleForTesting
-    static ApprovalStatus resolveApprovalStatus(@Nullable String approvalStatusLabel) {
-        if (approvalStatusLabel == null) {
+    static EvidenceLevelDetails resolveEvidenceLevelDetails(@Nullable String evidenceLevelDetails) {
+        if (evidenceLevelDetails == null) {
             return null;
         }
-        ApprovalStatus approvalStatus = ApprovalStatus.fromString(approvalStatusLabel);
-        if (approvalStatus == null) {
-            LOGGER.warn("Could not find enum value for approval status {}", approvalStatusLabel);
+
+        evidenceLevelDetails = evidenceLevelDetails.toLowerCase();
+        if (evidenceLevelDetails.contains("preclinical")) {
+            return EvidenceLevelDetails.PRECLINICAL.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("case report")) {
+            return EvidenceLevelDetails.CASE_REPORTS_SERIES.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("clinical study") || evidenceLevelDetails.contains("phase")) {
+            return EvidenceLevelDetails.CLINICAL_STUDY.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("guideline")) {
+            return EvidenceLevelDetails.GUIDELINE.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("fda approved")) {
+            return EvidenceLevelDetails.FDA_APPROVED.evidenceLevelDetailsCreator(evidenceLevelDetails);
+        }
+        else if (evidenceLevelDetails.contains("fda contraindicated")) {
+            return EvidenceLevelDetails.FDA_CONTRAINDICATED.evidenceLevelDetailsCreator(evidenceLevelDetails);
```

Comment:
This is incorrect and dodgy. You should not overwrite properties of an enum that should be private.

If you need to retain the source approval status you should add that as a separate field.